### PR TITLE
fix(scheduler): fix resource calculation way

### PR DESCRIPTION
**Pull request by shinytang6**
#1036  has been around for a long time and should be fixed asap.
Signed-off-by: shinytang6 <1074461480@qq.com>

### CODE REVIEW

File: pkg/scheduler/api/resource_info.go [hunk 1]
```diff
@@ -234,57 +234,72 @@ func (r *Resource) Multi(ratio float64) *Resource {
...
-		return l < r
+	cmpFunc := func(l, r float64) int {
+		lf := new(big.Rat).SetFloat64(l)
+		rf := new(big.Rat).SetFloat64(r)
+		return lf.Cmp(rf)
 	}
 
-	if !lessFunc(r.MilliCPU, rr.MilliCPU) {
+	isEqual := true
+	cpuCmp := cmpFunc(r.MilliCPU, rr.MilliCPU)
```
Comment (k82cn) [inline pkg/scheduler/api/resource_info.go:244]: I think we can update MilliCPU, memory to big.Rat

**Comment by k82cn** [comment]
> /hold

Why hold this PR?

File: pkg/scheduler/api/resource_info.go (hunk 1 above)
Comment (shinytang6) [reply pkg/scheduler/api/resource_info.go]: ok done, PTAL )

**Comment by Thor-wl** [comment]
Please resolve the conflicts

**Comment by Thor-wl** [comment]
Any process about this PR?

**Comment by shinytang6** [comment]
> Any process about this PR?

I think it's a big change to update all the cpu & memory to big.Rat. But my first commit in this PR can solve the logic problem, maybe we can accept that first?

File: pkg/scheduler/api/resource_info.go [hunk 2]
```diff
@@ -295,7 +310,10 @@ func (r *Resource) LessEqualStrict(rr *Resource) bool {
 // LessEqual checks whether a resource is less than other resource
```
Comment (Thor-wl) [inline pkg/scheduler/api/resource_info.go:310]: Please help add definition here, too

**Comment by Thor-wl** [comment]
Let's add some performance test about big.rat. 
/cc @Thor-wl @shinytang6

**Comment by Thor-wl** [comment]
The modification(https://github.com/volcano-sh/volcano/pull/1490) can be merged into this PR as well. @shinytang6

**Comment by shinytang6** [comment]
ref: https://github.com/volcano-sh/volcano/pull/1573
/close

**Pull request closed without merging**